### ISSUE TITLE
Feature: Impute initial concentrations for each subject at time zero (C0) 

### ISSUE THREAD

**Issue by Gero1999** [pharmaverse/aNCA#2]
**Issue description**
Sample records are not taken for obvious reasons exactly when the dose is provided. It is very standard to impute these values for each subject using different strategies depending on the route of administration, the time-close collected samples and the observed kinetics of the drug. 

**Notes**
1) If a drug is not IV bolus, C0=0. 
 2) If is IV bolus and the concentration decline is monoexponential it will be extrapolated using a linear regression
 3) If there are postdoses in the first five minutes, the closes to t0 can be used for C0.

**Comment by billdenney** [comment]
Related to #125

This is handled within PKNCA with the `impute` argument (see https://billdenney.github.io/pknca/articles/v08-data-imputation.html for details). Hopefully, you don't need to implement the imputation itself and only need to add the UI to use the existing imputation functions.

**Comment by Gero1999** [comment]
Yes, we are currently following the `impute` strategy. Although we are addressing some challenges on it for the App: 

**1. IV bolus AUC from 0**. I know that there is a specific parameter in PKNCA that you created for this (auc.iv) but we would like that still the user is allowed to make himself the proper AUC in the same parameter through imputation and directly for all AUC derived parameters. So we created this [IV bolus imputation methods:](https://github.com/pharmaverse/aNCA/blob/bug/auc-start-0/R/PKNCA_impute_method_additions.R) start_log, that uses pk.calc.c0, and when no possible then start_c1

**2. Certain parameters should never use imputed data**. Variables such as `tmax` or `cmax` should be by default be based only on observations. The way we are currently addressing this is by creating specific intervals for them that start from first observation.

**Comment by billdenney** [comment]
I'm not certain that I understand what you mean for point 1. In general, when calculating AUC for an IV bolus dose, you should have the percent back-extrapolated calculated to ensure that it's not extreme. That's why `auc.iv` exists and it's not the same as the other AUC methods.

For point 2, that should be handled by the way that PKNCA does calculations currently. If it doesn't only work based on observed data only for those parameters, that is a bug in PKNCA that should be fixed.

**Comment by Gero1999** [comment]
Let me clarify and maybe we can talk more about it perhaps in person tomorrow:

1. My understand is that for obtaining whole AUCs with back-extrapolations in regular cases (iv.bolus, extravascular/infusions...) you have two options: 
> 1)  Use the specific parameters for IV bolus (`auc.iv`) and extravascular (`auc.int`) that already consider imputed starts
> 2) Include an imputation PKNCA_start method to calculate correctly `auc` parameter with the imputation you want to do. 

Of course, the advantage I see for the second method is that it seems more flexible for different cases and concurrently updates all parameters that are related to the parameter `auc`. 


2.  I do not manage to find in the code how this is handled, but we can discuss it if not tomorrow. I am interested in understanding what would happen in the next scenario:

> If I have an iv bolus study and I compute an interval starting at the first observation (C1, Inf) with an imputation PKNCA_start method, will my `tmax` ignore the first value even if it was not really imputed?
